src/CredentialProvider.ts

```diff
@@ -3,15 +3,15 @@ import { Credentials } from './Credentials.ts'
 export class CredentialProvider {
   private credentials: Credentials
 
-  constructor({ accessKey, secretKey, sessionToken }: { accessKey: string; secretKey: string; sessionToken: string }) {
+  constructor({ accessKey, secretKey, sessionToken }: { accessKey: string; secretKey: string; sessionToken?: string }) {
     this.credentials = new Credentials({
       accessKey,
       secretKey,
       sessionToken,
     })
   }
 
-  async getCredentials(): Promise<Credentials | null> {
+  async getCredentials(): Promise<Credentials> {
```

Comment:
Should we put a comment here (as documentation, using JSDoc of course) about the breaking change of it not returning null?

Reply:
No, this is what I mentioned before, return a `null` will cause `Client` throw unhandled promise rejection. So `Client` already expect this method to return a `Credentials`.

https://github.com/minio/minio-js/blob/613932c40a958e0df02dd20d7dd50de1d0a88621/src/minio.js#L3474-L3481

Reply:
So it's not a breaking change afterall?

Reply:
Yes, consider this is the first time this class get typed.

Reply:
and also this is what I mentioned before, the ignored `unhandledRejection`

`Client().checkAndRefreshCreds` is never awaited

Reply:
Oh, it's fine then.